x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/embeddings/CohereEmbeddingsTaskSettings.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.cohere.embeddings;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.inference.InputType;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.TaskSettings;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.cohere.CohereTruncation;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalEnum;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalString;
+import static org.elasticsearch.xpack.inference.services.cohere.CohereServiceFields.MODEL;
+import static org.elasticsearch.xpack.inference.services.cohere.CohereServiceFields.TRUNCATE;
+
+/**
+ * Defines the task settings for the cohere text embeddings service.
+ *
+ * <p>
+ * <a href="https://docs.cohere.com/reference/embed">See api docs for details.</a>
+ * </p>
+ *
+ * @param model the id of the model to use in the requests to cohere
+ * @param inputType Specifies the type of input you're giving to the model
+ * @param embeddingType Specifies the type of embeddings you want to get back (we only support retrieving a single type)
+ * @param truncation Specifies how the API will handle inputs longer than the maximum token length
+ */
+public record CohereEmbeddingsTaskSettings(
+    @Nullable String model,
```

Comment:
I wonder if model belongs in Task Settings or Service Settings. 

The idea of task settings is that they can be overridden per request, but changing the model would produce incompatible results. The same argument can be applied to embedding type as the results would be incompatible with previously indexed results.

`OpenAiEmbeddingsTaskSettings` has set the precedent for model name in task settings but it is worth reviewing the idea.

Reply:
Yeah good point. I'll move them to the service settings specific to text embeddings. If we wanted to move them out of the task settings for openai, what would be the process there? Maybe the easiest thing to do would be leave them in the task settings but also add them to the service settings and let the service settings take priority?

Reply:
> leave them in the task settings but also add them to the service settings and let the service settings take priority?

It might be possible to automatically update the config to do this